Dart/src/com/jetbrains/lang/dart/ide/actions/DartPubActionBase.java

```diff
@@ -178,6 +178,8 @@ private static void doPerformPubAction(@NotNull final Module module,
 
     try {
       if (ourInProgress.compareAndSet(false, true)) {
+        command.withEnvironment("PUB_ENVIRONMENT", "IntelliJ-IDEA");
```

Comment:
More specific is good – assuming it's not a huge set of things.

This will help us understand what tools are being used.